### PR TITLE
feat(os): add mqx for osal

### PR DESCRIPTION
**Pull request by josepho0918**
### Description of the feature or fix

Add support for MQX RTOS.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/osal/lv_mqx.c [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_result_t lv_thread_init(lv_thread_t * thread, lv_thread_prio_t prio, void (*callback)(void *), size_t stack_size,
+                           void * user_data)
+{
+    TASK_TEMPLATE_STRUCT task_template;
+
+    memset(&task_template, 0, sizeof(task_template));
```
Comment (kisvegabor) [inline src/osal/lv_mqx.c]:
```suggestion
    lv_memset(&task_template, 0, sizeof(task_template));
```

**Review by kisvegabor — COMMENTED** [review]
Wao, it's great that we will have MQX support too! Thank you! 

May I ask for which chip do you use MQX and LVGL?

Would you mind adding a docs page [here](https://docs.lvgl.io/master/integration/os/index.html) for MQX? Nothing too complicated, just a brief introduction highlighting the benefits.

**Comment by josepho0918** [comment]
> Wao, it's great that we will have MQX support too! Thank you!
> 
> May I ask for which chip do you use MQX and LVGL?
> 
> Would you mind adding a docs page [here](https://docs.lvgl.io/master/integration/os/index.html) for MQX? Nothing too complicated, just a brief introduction highlighting the benefits.

I'm using MQX and LVGL on NXP i.MX RT1170.

File: src/osal/lv_mqx.c [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+lv_result_t lv_thread_sync_signal(lv_thread_sync_t * sync)
+{
+    _mqx_uint ret = _lwsem_post(&sync->sem);
+    if(ret != MQX_OK) {
+        LV_LOG_WARN("Error: %x", ret);
+        return LV_RESULT_INVALID;
+    }
+    else {
```
Comment (FASTSHIFT) [inline src/osal/lv_mqx.c]: remove "else"

File: src/osal/lv_mqx.c (hunk 1 above)
Comment (FASTSHIFT) [reply src/osal/lv_mqx.c]: Using `lv_memzero` is more concise~

**Comment by kisvegabor** [comment]
Adding the link in the docs is a good starting point. Thank you!

Please change `memset` to `lv_memzero` and we are good to go :slightly_smiling_face:

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]

**Comment by kisvegabor** [comment]
Thank you! I've just [shared it on LinkedIn](https://www.linkedin.com/posts/g%C3%A1bor-kiss-v%C3%A1mosi-4a3433154_mqx-support-was-just-added-to-lvgl-mqx-is-activity-7194429573533417472-tdn9?utm_source=share&utm_medium=member_desktop).

**Pull request merged**